diff -ur tcpdump-4.0.0/Makefile.in tcpdump-4.0.0-patched/Makefile.in
--- tcpdump-4.0.0/Makefile.in	2008-09-01 03:44:11.000000000 +0100
+++ tcpdump-4.0.0-patched/Makefile.in	2009-05-05 14:58:25.000000000 +0100
@@ -62,9 +62,9 @@
 # Explicitly define compilation rule since SunOS 4's make doesn't like gcc.
 # Also, gcc does not remove the .o before forking 'as', which can be a
 # problem if you don't own the file but can write to the directory.
-.c.o:
-	@rm -f $@
-	$(CC) $(CFLAGS) -c $(srcdir)/$*.c
+# .c.o:
+# 	@rm -f $@
+# 	$(CC) $(CFLAGS) -c $(srcdir)/$*.c
 
 CSRC =	addrtoname.c af.c checksum.c cpack.c gmpls.c oui.c gmt2local.c ipproto.c \
         nlpid.c l2vpn.c machdep.c parsenfsfh.c \
diff -ur tcpdump-4.0.0/print-enc.c tcpdump-4.0.0-patched/print-enc.c
--- tcpdump-4.0.0/print-enc.c	2008-09-01 03:44:22.000000000 +0100
+++ tcpdump-4.0.0-patched/print-enc.c	2009-05-05 15:04:17.000000000 +0100
@@ -77,9 +77,11 @@
 	case AF_INET:
 		ip_print(gndo, p, length);
 		break;
+#ifdef INET6
 	case AF_INET6:
 		ip6_print(p, length);
 		break;
+#endif
 	}
 
 out:
