diff -ur procServ-2.5.1/procServ.cc procServ-2.5.1-patched/procServ.cc
--- procServ-2.5.1/procServ.cc	2010-03-23 21:13:14.000000000 +0000
+++ procServ-2.5.1-patched/procServ.cc	2011-08-23 16:19:24.000000000 +0100
@@ -42,7 +42,7 @@
 char   *ignChars = NULL;         // Characters to ignore
 char   killChar = 0x18;          // Kill command character (default: ^X)
 char   toggleRestartChar = 0x14; // Toggle autorestart character (default: ^T)
-char   restartChar = 0x12;       // Restart character (default: ^R)
+char   restartChar = 0x18;       // Restart character (default: ^X)
 char   quitChar = 0x11;          // Quit character (default: ^Q)
 int    killSig = SIGKILL;        // Kill signal (default: SIGKILL)
 rlim_t coreSize;                 // Max core size for child
